### ISSUE TITLE
Not reporting errors from websocket route callback

### ISSUE THREAD

**Issue by kbadk** [HenningM/express-ws#87]
If an error occurs in the websocket route callback, the system fails silently. Using the [full example](https://github.com/HenningM/express-ws#full-example) (with 3 added `console.log`s):

```javascript
var express = require('express');
var app = express();
var expressWs = require('express-ws')(app);

app.use(function (req, res, next) {
  console.log('middleware');
  req.testing = 'testing';
  return next();
});

app.get('/', function(req, res, next) {
  console.log(SOMETHING_UNDEFINED_A);
  console.log('get route', req.testing);
  res.end();
});

app.ws('/', function(ws, req) {
  console.log(SOMETHING_UNDEFINED_B);

  ws.on('message', function(msg) {
    console.log(SOMETHING_UNDEFINED_C);
    console.log(msg);
  });
  console.log('socket', req.testing);
});

app.listen(3000);
```

We notice that navigating to http://localhost:3000/ correctly reports `ReferenceError: SOMETHING_UNDEFINED_A is not defined`.

Connecting to ws://localhost:3000/ on the other hand just fails silently. No error is reported.

If we remove `console.log(SOMETHING_UNDEFINED_B);` and connect and send a message to the server, `ReferenceError: SOMETHING_UNDEFINED_C is not defined` is also correctly reported.

To summarize: Errors occuring in the initial websocket route callback do not get reported.

**Comment by kbadk** [comment]
I've found that I can get the errors by doing

```javascript
app.use((err, req, res, next) => {
  console.error(err);
});
```

Not the solution I had hoped for, but it works. Would make more sense if an error was thrown in my opinion.

**Comment by andrewwakeling** [comment]
Worth noting, that this custom error-handling middleware needs to be defined last (given middleware is executed in the order it is defined).

It would be great to fix this library so it can leverage the default error handling that is in express.

In the very least, this behaviour should be documented in the README.md. This really caught me off-guard. Silent errors are a programmer's worst nightmare.
